bitcoin-core-dev-tech/2023-09/signature-aggregation.md

```diff
@@ -0,0 +1,148 @@
+
+In musig and frost you simply use schnorr verification (verify(sig, message, pubkey) => true/false) while sigagg verification is verify(sig, [(m1, p1), ….])
+
+## Benefits of halfagg
+
+More tx in a block, space savings
+
+But not much computational savings
+
+Coinjoin benefits (halfagg and fullagg)
+
+Concerns about fullagg actually hurting privacy
+
+Having a single signature via full-agg is easier to do on your own txs rather than on multi-party tx where interactivity is needed. This means privacy busting services like chainalysis can use that as a heuristic, assuming aggregated-sigs are a sign that all the inputs belong to the same person.
+
+Batch validation and Half-agg have same computation performance benefits
+
+Approach: Consensus change, v2 witness
+
+Half-agg consensus changes more invasive than taproot
+
+## Security proofs
+
+Under certain assumptions, if you can forge schnorr sig, you break discrete log
+
+Security loss of proof details
+
+When we say we have a security proof, there is nuance; under certain assumptions we prove that if we can forge a schnorr sig, you can also break discrete log. Most traditional Schnorr proof, the statement is of the form, if you have a probability x of forging a sign, that turns into a prob of some function of that (x^2) of breaking the discrete log.
+
+If we assume that the probability for computing the discreet log is not better than 2^128, this only implies that schnorr sig forgery is no more secure than 2^64. Which is not something that we would ordinarily accept. Crypto world has accepted this, theoretical world not so. Can you make your schnoor group big enough this is not a problem?
+
+Can target a certain security level by increasing group size, e.g. 512 bit group bight give 128 bit security.
+
+In practice world accepted that these loss factors are a side-effect of the proof, and not real. And that it actually is hard to break Schnorr signatures.
+
+There is an alt proof with different assumptions re. group size and hash size (the two params). The first one says we have a ^2 one for the group, but this one says use bigger hashes and make the group a bit smaller. Does this point to a lack of a fully-sensible proof? Using random-oracle stuff you might not get a tight proof.
+
+If you don't assume special things about the group, idealised assumptions, you have a loss of that square property. Someone has proved that there is no better proof. This doesn't imply it's insecure, but there is no better proof. In MuSig 1 paper there was a ^4 root loss. Similar here for half agg; there is a fork per signature, so there is a loss factor which is 2^n where n is number of transactions. So with 7 tx it only proves that signature is 1 bit secure.
+
+There is no perfect security proof, but in practice that is often tollerated/acceptable. But for half agg, since it's new, makes sense to be cautious.
```

Comment:
```suggestion
There is no perfect security proof, but in practice that is often tolerated/acceptable. But for half agg, since it's new, makes sense to be cautious.
```